algorithmic_efficiency/workloads/fastmri/fastmri_jax/workload.py

```diff
@@ -0,0 +1,288 @@
+  vy = cov_norm * (uyy - uy * uy)
+  vxy = cov_norm * (uxy - ux * uy)
+
+  c1 = (k1 * data_range)**2
+  c2 = (k2 * data_range)**2
+
+  a1 = 2 * ux * uy + c1
+  a2 = 2 * vxy + c2
+  b1 = ux**2 + uy**2 + c1
+  b2 = vx + vy + c2
+
+  d = b1 * b2
+  s = (a1 * a2) / d
+
+  # to avoid edge effects will ignore filter radius strip around edges
+  pad = (win_size - 1) // 2
+
+  # compute (weighted) mean of ssim.
+  return jnp.mean(s.at[pad:-pad, pad:-pad].get())
+
+
+def ssim(logits, targets, mean=None, std=None, volume_max=None):
+  """Computes example-wise structural similarity for a batch.
+
+  NOTE(dsuo): we use the same (default) arguments to `structural_similarity`
+  as in https://arxiv.org/abs/1811.08839.
+
+  Args:
+   logits: (batch,) + input.shape float array.
+   targets: (batch,) + input.shape float array.
+   mean: (batch,) mean of original images.
+   std: (batch,) std of original images.
+   volume_max: (batch,) of the volume max for the volumes each example came
+    from.
+  Returns:
+    Structural similarity computed per example, shape [batch, ...].
+  """
+  if volume_max is None:
+    volume_max = jnp.ones(logits.shape[0])
+
+  # NOTE(dsuo): `volume_max` can be 0 if we have a padded batch, but this will
+  # lead to NaN values in `ssim`.
+  volume_max = jnp.where(volume_max == 0, jnp.ones_like(volume_max), volume_max)
+
+  if mean is None:
+    mean = jnp.zeros(logits.shape[0])
+
+  if std is None:
+    std = jnp.ones(logits.shape[0])
+
+  mean = mean.reshape((-1,) + (1,) * (len(logits.shape) - 1))
+  std = std.reshape((-1,) + (1,) * (len(logits.shape) - 1))
+  logits = logits * std + mean
+  targets = targets * std + mean
+  print(logits.shape, targets.shape, volume_max.shape)
+  ssims = jax.vmap(structural_similarity)(logits, targets, volume_max)
+  return ssims
+
+
+class FastMRIWorkload(BaseFastMRIWorkload):
+
+  def __init__(self):
+    super().__init__()
+    self._param_types = None
+    self._eval_iters = {}
+    self._model = models.UNet()
+
+  @property
+  def param_shapes(self):
+    if self._param_shapes is None:
+      raise ValueError(
+          'This should not happen, workload.init_model_fn() should be called '
+          'before workload.param_shapes!')
+    return self._param_shapes
+
+  @property
+  def model_params_types(self):
+    """The shapes of the parameters in the workload model."""
+    if self._param_types is None:
+      self._param_types = param_utils.pytorch_param_types(self._param_shapes)
+    return self._param_types
+
+  def build_input_queue(self,
+                        data_rng,
+                        split: str,
+                        data_dir: str,
+                        global_batch_size: int,
+                        cache: Optional[bool] = None,
+                        repeat_final_dataset: Optional[bool] = None,
+                        num_batches: Optional[int] = None):
+    del cache
+    per_host_batch_size = global_batch_size // jax.num_local_devices()
```

Comment:
As I mentioned in my comments in `load_fastmri_split()`, it seems like `global_batch_size` should be passed on.